jiant/trainer.py

```diff
@@ -591,31 +605,29 @@ def clip_function(grad):
             n_batches_since_val = task_info["n_batches_since_val"]
             tr_loss = task_info["loss"]
 
-            for batch in itertools.islice(tr_generator, 1):
-                n_batches_since_val += 1
-                total_batches_trained += 1
-                optimizer.zero_grad()
+            for batch in itertools.islice(tr_generator, self._accumulation_steps):
                 output_dict = self._forward(batch, task=task)
-                assert_for_log(
-                    "loss" in output_dict, "Model must return a dict containing a 'loss' key"
-                )
-                loss = get_output_attribute(
-                    output_dict, "loss", self._cuda_device
-                )  # optionally scale loss
+                assert_for_log("loss" in output_dict, "Model must return a dict with 'loss' key")
+                loss = get_output_attribute(output_dict, "loss", self._cuda_device)
+                if self._accumulation_steps > 1:
+                    loss = loss / self._accumulation_steps
```

Comment:
Double check this. I could be wrong, but I thought that the loss functions we use all _sum_ the loss over entries in a batch. In that case, if we're trying to simulate a larger batch size, we should sum over these sub-batches, not average.

Reply:
The loss functions I tracked down are using the default mean for reduction, so I think this is OK.